python/vosk/__init__.py

```diff
@@ -47,15 +47,15 @@ def __del__(self):
 
 class KaldiRecognizer(object):
 
-    def __init__(self, *args):
-        if len(args) == 2:
+    def __init__(self, stream, sample_rate, model=None):
+        if model is None:
             self._handle = _c.vosk_recognizer_new(args[0]._handle, args[1])
-        elif len(args) == 3 and type(args[2]) is SpkModel:
-            self._handle = _c.vosk_recognizer_new_spk(args[0]._handle, args[1], args[2]._handle)
-        elif len(args) == 3 and type(args[2]) is str:
+        elif isinstance(model, str):
             self._handle = _c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))
```

Comment:
You removed args above, is it going to work at all?

Reply:
Yes. You can test it if you want.

Reply:
hello?

Reply:
Hey :)
Wouldn't you want to use the keyword arguments for the sake of clarity?
e.g. `_c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))` => `_c.vosk_recognizer_new(stream._handle, sample_rate, model.encode('utf-8'))`

Reply:
that's what my PR does...

Reply:
What I meant is:
In the PR the __init__ signature has keyword arguments, but when calling other functions in its body, args is used instead.

Current code in the PR:
```
def __init__(self, stream, sample_rate, model=None):
        if model is None:
            self._handle = _c.vosk_recognizer_new(args[0]._handle, args[1])
        elif isinstance(model, str):
            self._handle = _c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))
        elif isinstance(model, SpkModel):
            self._handle = _c.vosk_recognizer_new_spk(args[0]._handle, args[1], args[2]._handle)
        else:
            raise TypeError("model argument must be either None, str or SpkModel")

        if self._handle == _ffi.NULL:
            raise Exception("Failed to create a recognizer")
```

Could be:
```
    def __init__(self, stream, sample_rate, model=None):
        if model is None:
            self._handle = _c.vosk_recognizer_new(stream._handle, sample_rate)
        elif isinstance(model, str):
            self._handle = _c.vosk_recognizer_new_grm(stream._handle, sample_rate, model.encode('utf-8'))
        elif isinstance(model, SpkModel):
            self._handle = _c.vosk_recognizer_new_spk(stream._handle, sample_rate, model._handle)
        else:
            raise TypeError("model argument must be either None, str or SpkModel")

        if self._handle == _ffi.NULL:
            raise Exception("Failed to create a recognizer")
```

Reply:
My bad...
Currently I am tired. So please send a PR to my branch and fork or wait for me to fix it.